contracts/Polygon_SpokePool.sol

```diff
@@ -127,6 +127,16 @@ contract Polygon_SpokePool is IFxMessageProcessor, SpokePool {
         require(success, "delegatecall failed");
     }
 
+    /**
+     * @notice Allows the caller to trigger the wrapping of any unwrapped matic tokens.
+     * @dev Matic sends via L1 -> L2 bridging actions don't call into the contract receiving the tokens, so wrapping
+     * must be done via a separate transaction.
+     */
+    function wrap() public nonReentrant {
```

Comment:
Should `wrap` be triggered on any other external functions like `processMessageFromRoot` for convenience?

Reply:
It could. Note: the functions that actually would need the tokens aren't triggered by processMessageFromRoot, though. executing relayer refunds and slow relays are the ones that would require this, but to add it to those, they might need to be overridden.

Will think a little about how to do this with little overhead.

Reply:
Added this functionality in the last commit: https://github.com/across-protocol/contracts-v2/pull/128/commits/811ac20674d28189fd01297c05ce5b9e89f7a183. PTAL.

Reply:
I don't really think this extra check is worthwhile because:
1. We already check `address(this).balance > 0`, which is _extremely_ cheap. It costs 5 gas according to https://ethereum.org/en/developers/docs/evm/opcodes/ to read the balance. Reading the `weth` value from the contract, by comparison, would cost 100 - 2100 gas. I think these checks are the only difference between the two, since we will likely run _some_ matic execution step each time matic tokens are sent over to the spoke, meaning the number of wraps shouldn't really differ between the two.
2. Since this is on polygon, gas costs shouldn't really matter too much.

Reply:
As discussed offline, we'll leave this as is.